client/alloc_runner.go

```diff
@@ -119,16 +119,17 @@ func (r *AllocRunner) RestoreState() error {
 
 	// Restore fields
 	r.alloc = snap.Alloc
-	r.ctx = snap.Context
+	r.allocDir = snap.AllocDir
 	r.allocClientStatus = snap.AllocClientStatus
 	r.allocClientDescription = snap.AllocClientDescription
 
 	var snapshotErrors multierror.Error
 	if r.alloc == nil {
 		snapshotErrors.Errors = append(snapshotErrors.Errors, fmt.Errorf("alloc_runner snapshot includes a nil allocation"))
 	}
-	if r.ctx == nil {
-		snapshotErrors.Errors = append(snapshotErrors.Errors, fmt.Errorf("alloc_runner snapshot includes a nil context"))
+	if r.allocDir == nil {
+		//FIXME Upgrade path?
```

Comment:
You could probably keep the context for now, and if the `r.allocDir` is still nil, create it. That should take care of the upgrade scenario. And remove context in the next release.